### ISSUE TITLE
sbnanaobj Does Not Install .pcm File for StandardRecord Dictionary

### ISSUE THREAD

**Issue by gputnam** [SBNSoftware/sbnanaobj#4]
This is a regression from the restructuring. 

The .pcm file is needed to run CAF-maker. Currently running CAF-maker will crash with a complaint that it can't find the caf::StandardRecord dictionary. This is because the .pcm file is not correctly being installed.

It looks like the regression was caused by removing the lines:

include(ArtDictionary)
include(ArtMake)
include(BuildPlugins)

From the top-level CMakeLists.txt when moving from sbncode to sbnanaobj.

**Comment by cjbackhouse** [comment]
We don't want sbnanaobj to depend on art. That means I think those cmake lines are unavailable. Do we know how to install the pcm file without using them? We are using cet for the dictionary. Is there another cet command we need, or is this a bug/oversight in the existing command?

Thanks for all the testing and sorry for the oversights.

**Comment by gputnam** [comment]
I don't know how to install the .pcm files outside of the art CMake macros. You could make the .pcm files a custom target and install them. I don't know if cet or ROOT provide alternate macros than the one being used.

**Comment by cjbackhouse** [comment]
I am looking at
https://cdcvs.fnal.gov/redmine/projects/cetbuildtools/repository/revisions/master/entry/Modules/BuildDictionary.cmake

and there is logic for rootmap and pcm files. You don't have to opt-in, looks like it should happen automatically for root6.

I _do_ see `build_slf7.x86_64/sbnanaobj/lib/libsbnanaobj_StandardRecord_dict_rdict.pcm` get generated, but it doesn't appear to make it to my localProducts directory

**Comment by gputnam** [comment]
I see, I guess that makes sense. 

The problem then is that mrb configures ROOT to look at localProducts for dictionaries, not the build area. So I think one would need to add a command to install the .pcm file in the additional place -- or maybe there is some way to overwrite the default installation location in the function?

**Comment by cjbackhouse** [comment]
I'm working on the hypothesis that either this is a bug in cetbuildtools, or that we're supposed to call an additional install_dicts kind of command, but I am not an expert on mrb, cmake, or cet, so I'm mostly floundering.

**Comment by cjbackhouse** [comment]
The existing makefile in sbncode before the split just called `build_dictionary`. From searching in the products in cvmfs it looks like sbncode v09_14_00 did indeed have a lib/libStandardRecord_dict_rdict.pcm. At least in my checkout I'm not creating a pcm file at all, not even in the build directory.
The current sbnanaobj/StandardRecord/CMakeLists.txt is really extremely similar to the previous sbncode/StandardRecord/CMakeLists.txt

**Comment by cjbackhouse** [comment]
Fixed with https://github.com/SBNSoftware/sbnanaobj/commit/71efdd3c237c025395e825ab60a63b40bdfea5cc

We needed to include BuildDictionary explicitly. I think what must be happening is that we otherwise pick up some other, less featured, version of build_directory from somewhere else. That deserves further investigation.

It's worth noting that I still can't produce CAFs, because setting up sbndcode causes it to go looking for the dictionaries in the sbnanaobj external product. I think it would work if you added sbndcode to your mrb area, but this is kind of suboptimal.
